pkg/transportrequest/rfc/upload.go

```diff
@@ -0,0 +1,152 @@
+func (action *UploadAction) WithApplicationURL(z string) {
+	action.ApplicationURL = z
+}
+
+// WithTransportRequestID The transport request ID for the upload
+func (action *UploadAction) WithTransportRequestID(t string) {
+	action.TransportRequestID = t
+}
+
+// WithApplication Everything we need to know about the application
+func (action *UploadAction) WithApplication(a Application) {
+	action.Application = a
+}
+
+// WithConfiguration Everything we need to know in order to perform the upload
+func (action *UploadAction) WithConfiguration(c UploadConfig) {
+	action.Configuration = c
+}
+
+// WithConnection Everything we need to know about the connection
+func (action *UploadAction) WithConnection(c Connection) {
+	action.Connection = c
+}
+
+// Perform Performs the upload
+func (action *UploadAction) Perform(command Exec) error {
+
+	log.Entry().Infof("Deploying artifact '%s' to '%s', client: '%s', instance: '%s'.",
+		action.ApplicationURL, action.Connection.Endpoint, action.Connection.Client, action.Connection.Instance,
+	)
+
+	missingParameters, err := validation.FindEmptyStringsInConfigStruct(*action)
+	if err != nil {
+		return fmt.Errorf("Cannot check that all required parameters are available for SOLMAN upload: %w", err)
+	}
+	notInitialized := len(missingParameters) != 0
+	if notInitialized {
+		return fmt.Errorf("Cannot perform artifact upload. The following parameters are not available %s", missingParameters)
+	}
+
+	command.SetEnv([]string{
+		"ABAP_DEVELOPMENT_SERVER" + eq + action.Connection.Endpoint,
+		"ABAP_DEVELOPMENT_USER" + eq + action.Connection.User,
+		"ABAP_DEVELOPMENT_PASSWORD" + eq + action.Connection.Password,
+		"ABAP_DEVELOPMENT_INSTANCE" + eq + action.Connection.Instance,
+		"ABAP_DEVELOPMENT_CLIENT" + eq + action.Connection.Client,
+		"ABAP_APPLICATION_NAME" + eq + action.Application.Name,
+		"ABAP_APPLICATION_DESC" + eq + action.Application.Description,
+		"ABAP_PACKAGE" + eq + action.Application.AbapPackage,
+		"ZIP_FILE_URL" + eq + action.ApplicationURL,
+		"CODE_PAGE" + eq + action.Configuration.CodePage,
+		"ABAP_ACCEPT_UNIX_STYLE_EOL" + eq + toAbapBool(action.Configuration.AcceptUnixStyleEndOfLine),
+		"FAIL_UPLOAD_ON_WARNING" + eq + strconv.FormatBool(action.Configuration.FailUploadOnWarning),
+		"VERBOSE" + eq + strconv.FormatBool(action.Configuration.Verbose),
+	})
+
+	err = command.RunExecutable("cts", fmt.Sprintf("uploadToABAP:%s", action.TransportRequestID))
+
+	if err == nil {
+		exitCode := command.GetExitCode()
+		if exitCode != 0 {
+			err = fmt.Errorf("received return code not equal zero: '%d'. Check log for details", exitCode)
+		}
+	}
+
+	if err == nil {
+		log.Entry().Infof("Deploying artifact '%s' to '%s', client: '%s', instance: '%s' succeeded.",
+			action.ApplicationURL, action.Connection.Endpoint, action.Connection.Client, action.Connection.Instance,
+		)
+	} else {
+		log.Entry().Infof("Deploying artifact '%s' to '%s', client: '%s', instance: '%s' failed.",
```

Comment:
`log.Entry().WithError(err).Warnf` instead of `Infof`

Reply:
done